### DISCUSSION TITLE
Exception Unwrapping

### DISCUSSION THREAD

**Discussion by cliedeman** [getsentry/sentry-dotnet#3182] [General]
### Problem Statement

I have an Exception where I am interested in the inner exception. I would like to unwrap the exception and allow the inner exception to continue down the exception processing pipeline - filtering etc.

**Comment by jamescrosswell** [comment]
Hey @cliedeman, are you able to isolate the AggregateException via a catch and rethrow in your code?

```csharp
try
{
    throw new AggregateException(new Exception("This is a test exception"));;
}
catch (Exception e)
{
    if (e is AggregateException { InnerException: not null } ae)
    {
        SentrySdk.CaptureException(ae.InnerException);
    }
    else
    {
        SentrySdk.CaptureException(e);
    }
}
```

**Comment by cliedeman** [comment]
Hi @jamescrosswell 

I have tried something like this before. But then it doesn't pass through my aspnetcore sentry middleware.

**Comment by jamescrosswell** [comment]
> I have tried something like this before. But then it doesn't pass through my aspnetcore sentry middleware.

You could do this (that way it would get caught by the global exception handler logic from the middleware):

```csharp
try
{
    throw new AggregateException(new Exception("This is a test exception"));;
}
catch (Exception e)
{
    if (e is AggregateException { InnerException: not null } ae)
    {
        throw ae.InnerException;
    }
    else
    {
        throw;
    }
}
```

**Reply by cliedeman** [reply]
Thanks for the idea @jamescrosswell 
Testing an endpoint filter that does something close to that and will report back.
